daemon/monitor.go

```diff
@@ -39,9 +39,8 @@ func (daemon *Daemon) handleContainerExit(c *container.Container, e *libcontaine
 
 	tsk, ok := c.Task()
 	if ok {
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		ctx := context.Background()
```

Comment:
In this case we are possibly keeping the container mutex locked indefinitely which will end up breaking everything.

I think we need some way to synchronize this better.